### ISSUE TITLE
Exception #0 (ReflectionException): Class Magento\Catalog\ViewModel\Product\OptionsData does not exist

### ISSUE THREAD

**Issue by tim-breitenstein-it** [magesuite/brand-management#10]
`Exception #0 (ReflectionException): Class Magento\Catalog\ViewModel\Product\OptionsData does not exist` after update from v1.6.0 to v1.6.9.

- Magento: v2.4.2-p2
- Magesuite: v7.0.0

`/app/code/Magento/Catalog/ViewModel/Product/OptionsData.php` is only available in Magento v2.4.3:

- https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Catalog/ViewModel/Product/OptionsData.php
- https://github.com/magento/magento2/blob/2.4.2-p2/app/code/Magento/Catalog/ViewModel/Product/OptionsData.php

Detailed Log:
```
1 exception(s):
Exception #0 (ReflectionException): Class Magento\Catalog\ViewModel\Product\OptionsData does not exist

Exception #0 (ReflectionException): Class Magento\Catalog\ViewModel\Product\OptionsData does not exist
<pre>#1 Magento\Framework\Code\Reader\ClassReader->getConstructor() called at [vendor/magento/framework/ObjectManager/Definition/Runtime.php:54]
#2 Magento\Framework\ObjectManager\Definition\Runtime->getParameters() called at [vendor/magento/framework/ObjectManager/Factory/Compiled.php:100]
#3 Magento\Framework\ObjectManager\Factory\Compiled->create() called at [vendor/magento/framework/ObjectManager/ObjectManager.php:70]
#4 Magento\Framework\ObjectManager\ObjectManager->get() called at [vendor/magento/framework/View/Layout/Argument/Interpreter/DataObject.php:61]
#5 Magento\Framework\View\Layout\Argument\Interpreter\DataObject->evaluate() called at [vendor/magento/framework/Data/Argument/Interpreter/Composite.php:61]
#6 Magento\Framework\Data\Argument\Interpreter\Composite->evaluate() called at [vendor/magento/framework/View/Layout/Argument/Interpreter/Decorator/Updater.php:47]
#7 Magento\Framework\View\Layout\Argument\Interpreter\Decorator\Updater->evaluate() called at [vendor/magento/framework/View/Layout/Generator/Block.php:320]
#8 Magento\Framework\View\Layout\Generator\Block->evaluateArguments() called at [vendor/magento/framework/View/Layout/Generator/Block.php:229]
#9 Magento\Framework\View\Layout\Generator\Block->generateBlock() called at [vendor/magento/framework/View/Layout/Generator/Block.php:134]
#10 Magento\Framework\View\Layout\Generator\Block->process() called at [vendor/magento/framework/View/Layout/GeneratorPool.php:93]
#11 Magento\Framework\View\Layout\GeneratorPool->process() called at [vendor/magento/framework/View/Layout.php:365]
#12 Magento\Framework\View\Layout->generateElements() called at [vendor/magento/framework/Interception/Interceptor.php:58]
#13 Magento\Framework\View\Layout\Interceptor->___callParent() called at [vendor/magento/framework/Interception/Interceptor.php:138]
#14 Magento\Framework\View\Layout\Interceptor->Magento\Framework\Interception\{closure}() called at [vendor/magento/framework/Interception/Interceptor.php:153]
#15 Magento\Framework\View\Layout\Interceptor->___callPlugins() called at [generated/code/Magento/Framework/View/Layout/Interceptor.php:32]
#16 Magento\Framework\View\Layout\Interceptor->generateElements() called at [vendor/magento/framework/View/Layout/Builder.php:129]
#17 Magento\Framework\View\Layout\Builder->generateLayoutBlocks() called at [vendor/magento/framework/View/Page/Builder.php:65]
#18 Magento\Framework\View\Page\Builder->generateLayoutBlocks() called at [vendor/magento/framework/View/Layout/Builder.php:65]
#19 Magento\Framework\View\Layout\Builder->build() called at [vendor/magento/framework/View/Page/Config.php:224]
#20 Magento\Framework\View\Page\Config->build() called at [vendor/magento/framework/View/Page/Config.php:260]
#21 Magento\Framework\View\Page\Config->setMetadata() called at [vendor/magento/framework/View/Page/Config.php:447]
#22 Magento\Framework\View\Page\Config->setRobots() called at [vendor/creativestyle/magesuite-brand-management/Controller/Index/Index.php:77]
#23 MageSuite\BrandManagement\Controller\Index\Index->execute() called at [vendor/magento/framework/Interception/Interceptor.php:58]
#24 MageSuite\BrandManagement\Controller\Index\Index\Interceptor->___callParent() called at [vendor/magento/framework/Interception/Interceptor.php:138]
#25 MageSuite\BrandManagement\Controller\Index\Index\Interceptor->Magento\Framework\Interception\{closure}() called at [vendor/magento/framework/App/Action/Plugin/ActionFlagNoDispatchPlugin.php:51]
#26 Magento\Framework\App\Action\Plugin\ActionFlagNoDispatchPlugin->aroundExecute() called at [vendor/magento/framework/Interception/Interceptor.php:135]
#27 MageSuite\BrandManagement\Controller\Index\Index\Interceptor->Magento\Framework\Interception\{closure}() called at [vendor/magento/framework/Interception/Interceptor.php:153]
#28 MageSuite\BrandManagement\Controller\Index\Index\Interceptor->___callPlugins() called at [generated/code/MageSuite/BrandManagement/Controller/Index/Index/Interceptor.php:23]
#29 MageSuite\BrandManagement\Controller\Index\Index\Interceptor->execute() called at [vendor/magento/framework/App/Action/Action.php:111]
#30 Magento\Framework\App\Action\Action->dispatch() called at [vendor/magento/framework/Interception/Interceptor.php:58]
#31 MageSuite\BrandManagement\Controller\Index\Index\Interceptor->___callParent() called at [vendor/magento/framework/Interception/Interceptor.php:138]
#32 MageSuite\BrandManagement\Controller\Index\Index\Interceptor->Magento\Framework\Interception\{closure}() called at [vendor/magento/framework/Interception/Interceptor.php:153]
#33 MageSuite\BrandManagement\Controller\Index\Index\Interceptor->___callPlugins() called at [generated/code/MageSuite/BrandManagement/Controller/Index/Index/Interceptor.php:32]
#34 MageSuite\BrandManagement\Controller\Index\Index\Interceptor->dispatch() called at [vendor/magento/framework/App/FrontController.php:186]
#35 Magento\Framework\App\FrontController->processRequest() called at [vendor/magento/framework/App/FrontController.php:118]
#36 Magento\Framework\App\FrontController->dispatch() called at [vendor/magento/framework/Interception/Interceptor.php:58]
#37 Magento\Framework\App\FrontController\Interceptor->___callParent() called at [vendor/magento/framework/Interception/Interceptor.php:138]
#38 Magento\Framework\App\FrontController\Interceptor->Magento\Framework\Interception\{closure}() called at [vendor/magento/module-store/App/FrontController/Plugin/RequestPreprocessor.php:99]
#39 Magento\Store\App\FrontController\Plugin\RequestPreprocessor->aroundDispatch() called at [vendor/magento/framework/Interception/Interceptor.php:135]
#40 Magento\Framework\App\FrontController\Interceptor->Magento\Framework\Interception\{closure}() called at [vendor/magento/module-page-cache/Model/App/FrontController/BuiltinPlugin.php:75]
#41 Magento\PageCache\Model\App\FrontController\BuiltinPlugin->aroundDispatch() called at [vendor/magento/framework/Interception/Interceptor.php:58]
#42 Magento\PageCache\Model\App\FrontController\BuiltinPlugin\Interceptor->___callParent() called at [vendor/magento/framework/Interception/Interceptor.php:138]
#43 Magento\PageCache\Model\App\FrontController\BuiltinPlugin\Interceptor->Magento\Framework\Interception\{closure}() called at [vendor/magento/framework/Interception/Interceptor.php:153]
#44 Magento\PageCache\Model\App\FrontController\BuiltinPlugin\Interceptor->___callPlugins() called at [generated/code/Magento/PageCache/Model/App/FrontController/BuiltinPlugin/Interceptor.php:23]
#45 Magento\PageCache\Model\App\FrontController\BuiltinPlugin\Interceptor->aroundDispatch() called at [vendor/magento/framework/Interception/Interceptor.php:135]
#46 Magento\Framework\App\FrontController\Interceptor->Magento\Framework\Interception\{closure}() called at [vendor/magento/framework/Interception/Interceptor.php:153]
#47 Magento\Framework\App\FrontController\Interceptor->___callPlugins() called at [generated/code/Magento/Framework/App/FrontController/Interceptor.php:23]
#48 Magento\Framework\App\FrontController\Interceptor->dispatch() called at [vendor/magento/framework/App/Http.php:116]
#49 Magento\Framework\App\Http->launch() called at [vendor/magento/framework/Interception/Interceptor.php:58]
#50 Magento\Framework\App\Http\Interceptor->___callParent() called at [vendor/magento/framework/Interception/Interceptor.php:138]
#51 Magento\Framework\App\Http\Interceptor->Magento\Framework\Interception\{closure}() called at [vendor/magento/framework/Interception/Interceptor.php:153]
#52 Magento\Framework\App\Http\Interceptor->___callPlugins() called at [generated/code/Magento/Framework/App/Http/Interceptor.php:23]
#53 Magento\Framework\App\Http\Interceptor->launch() called at [vendor/magento/framework/App/Bootstrap.php:263]
#54 Magento\Framework\App\Bootstrap->run() called at [pub/index.php:29]
</pre>
```

**Comment by diwipl** [comment]
Thank you very much for reporting this.

We're preparing a fix at the moment, it should be done by the end of the week.

**Comment by tim-breitenstein-it** [comment]
Solved by [v1.7.0](https://github.com/magesuite/brand-management/releases/tag/v1.7.0) https://github.com/magesuite/brand-management/commit/0701f0531649401a5f93ac4346005d9047005516
